Per product NPS survey

annikaschmid opened the pull request.

annikaschmid commented on requests-for-comments/2023-10-25-per-product-nps.md:
```diff
@@ -0,0 +1,47 @@
+
+  <img width="343" alt="Screenshot 2023-10-25 at 14 29 05" src="https://github.com/PostHog/meta/assets/14750837/e2388e9b-0eb5-4749-ad18-00ef3ee2c3f2">
+  
+- Thank you: Get a free PostHog sticker (optional):      
```
@corywatilo I know we do this for applicants. Would this be easy to set this up for surveys, and send users to a URL where they can get a sticker, that only works once? If not, we can also de-prioritise this

corywatilo replied:
Yep, or we can just use a shared code (like we do for applicants). When we're ready, hit me up and I can generate.

annikaschmid replied:
@corywatilo I am just waiting for a quick bug fix from the feature success team, then this is ready to go. If I could have this link by the end of the week / early next week, this would be great.

annikaschmid commented:
@mariusandra @benjackwhite @neilkakkar FYI, since your teams are responsible for these products

raquelmsmith commented on requests-for-comments/2023-10-25-per-product-nps.md:
```diff
@@ -0,0 +1,47 @@
+
+## Release conditions
+
+- Visitors on any feature flags page (app & eu). I am using the following regex: `https://(app|eu)\.posthog\.com/feature_flags`
```
can we make it so that they only get it if they signed up 30 days ago or something? we really don't want to ask for nps score if someone just signed up...

annikaschmid replied:
Good point, I think we can do this. I'll update the targeting

annikaschmid replied:
@raquelmsmith Since I've updated the targeting to only target activated users, I have removed the 'signed up before' condition

corywatilo commented on requests-for-comments/2023-10-25-per-product-nps.md:
```diff
@@ -0,0 +1,47 @@
+## Release conditions
+
+- Visitors on any feature flags page (app & eu). I am using the following regex: `https://(app|eu)\.posthog\.com/feature_flags`
+    - Note: Once we have behavioural surveys, we can update the targeting to only show to users who have created a flag / watched 10 recordings / created a survey / created an insight / created an experiment
```
Yep, seems like there are some good places to drop these in - when we know a customer has completed a significant action:

- Product analytics: Saved an insight, added an insight to a dashboard, etc
- Session replay: Shared a replay? etc
- Flag: Enabled a flag
- A/B test: Ended a test
- Survey: Launched a survey

Is the plan to launch this once we have behavioral surveys?

annikaschmid replied:
I was planning to launch this before we have behavioural surveys, because I don't want to wait.

That means, the targeting would be they are on the e.g. feature flags page AND have signed up at least 30 days ago. It will be less accurate than what we can do once we have behavioural surveys, but I think it's worth collecting feedback already, even if we know it will not be perfect.

annikaschmid replied:
I've set this up with cohorts. It's quite hacky, but it works.

annikaschmid commented on requests-for-comments/2023-10-25-per-product-nps.md:
```diff
@@ -0,0 +1,47 @@
+
+## Release conditions
+
+- Visitors on any feature flags page (app & eu). I am using the following regex: `https://(app|eu)\.posthog\.com/feature_flags`
```
```suggestion
- Visitors on any feature flags page (app & eu). I am using the following regex: `https://(app|eu)\.posthog\.com/feature_flags`
- Visitors who have signed up at least 30 days ago: `joined_at` before `date`
```

raquelmsmith commented:
>We should add a question which aligns with the role_at_organization property, asking users if they are engineering, founder, product, sales, marketing, or other.

shouldn't we know this already based on the info they provide at signup? (I know not everyone answers this question, but we can maybe draw conclusions from those who do)

> remove the 'last 14 days' targeting, as we can't easily target this correctly

I really think we don't want to ask people for NPS scores if they've just joined. Is there some way we can still do this delay?

joethreepwood commented:
> shouldn't we know this already based on the info they provide at signup? (I know not everyone answers this question, but we can maybe draw conclusions from those who do)

If someone knows a way to correlate this, sure. My main interest is: What roles are using which products, and how do we perform for them? I don't mind how we get that info. 

> I really think we don't want to ask people for NPS scores if they've just joined. Is there some way we can still do this delay?

I agree 100%, I just couldn't see a way to implement this delay. At the moment the survey is only showing to users who signed up before 2023-11-01, which also isn't ideal because we're not getting current info.

annikaschmid commented:
> I agree 100%, I just couldn't see a way to implement this delay. At the moment the survey is only showing to users who signed up before 2023-11-01, which also isn't ideal because we're not getting current info.

Yes, because we can't do < 14 days properly, I need to regularly update this date manually every couple of weeks. Which is not ideal, but I don't think there is a better solution right now. This is European format, so we are only slightly out of date. I will set myself a calendar reminder so that I won't forget.

> If someone knows a way to correlate this, sure. My main interest is: What roles are using which products, and how do we perform for them? I don't mind how we get that info.

I just had a look how we could get this. If you want to see this on the survey results page, here is how: https://posthog.slack.com/archives/C0351B1DMUY/p1702399749511089 
We have data for ~47% of users, which I think is ok as a start.

annikaschmid commented:
> lower the limit to users who have seen another survey to exclude users who've seen a survey in the last 14 days
> increase the rollout from the current 20% to 50% of users

Done. I've rolled this out according to your proposal, and also updated the signup date. I will have a look to see if this makes a dent in the response rate.

> Remove the free sticker

Done. I replaced it with a auto-disappearing "Thank you for your feedback!" message.

> Rollout to product analytics

Going to do this this week. I will also set this up for session replay.
